catris.py

```diff
@@ -555,77 +1006,43 @@ def __init__(self, client: Client, player: Player):
-                name_line += color_bytes
-
-                if player == self.player:
-                    header_line += b"==" * WIDTH_PER_PLAYER
-                else:
-                    header_line += b"--" * WIDTH_PER_PLAYER
-                name_line += display_name.center(2 * WIDTH_PER_PLAYER).encode("utf-8")
-
-            name_line += COLOR % 0
-            header_line += COLOR % 0
-            header_line += b"o"
-
-            lines = [name_line, header_line]
-
-            for blink_y, row in enumerate(state.get_square_colors()):
-                line = b"|"
-                for color in row:
-                    if color is None:
-                        line += b"  "
-                    else:
-                        line += COLOR % color
-                        line += b"  "
-                        line += COLOR % 0
-                line += b"|"
-                lines.append(line)
-
-            lines.append(b"o" + b"--" * state.get_width() + b"o")
-
+        with self._client.server.access_game(render=False) as state:
+            lines = state.get_lines_to_render(self.player)
             lines[5] += f"  Score: {state.score}".encode("ascii")
             if self._client.rotate_counter_clockwise:
                 lines[6] += b"  Counter-clockwise"
             if isinstance(self.player.moving_block_or_wait_counter, int):
                 n = self.player.moving_block_or_wait_counter
                 lines[8] += f"  Please wait: {n}".encode("ascii")
-
             return lines
 
     def handle_key_press(self, received: bytes) -> None:
         if received in (b"A", b"a", LEFT_ARROW_KEY):
-            with self._client.server.access_game_state() as state:
-                state.move_if_possible(self.player, dx=-1, dy=0)
+            with self._client.server.access_game() as state:
+                state.move_if_possible(self.player, dx=-1, dy=0, in_player_coords=True)
         elif received in (b"D", b"d", RIGHT_ARROW_KEY):
-            with self._client.server.access_game_state() as state:
-                state.move_if_possible(self.player, dx=1, dy=0)
+            with self._client.server.access_game() as state:
+                state.move_if_possible(self.player, dx=1, dy=0, in_player_coords=True)
         elif received in (b"W", b"w", UP_ARROW_KEY, b"\r"):
-            with self._client.server.access_game_state() as state:
+            with self._client.server.access_game() as state:
                 state.rotate(self.player, self._client.rotate_counter_clockwise)
         elif received in (b"S", b"s", DOWN_ARROW_KEY, b" "):
-            with self._client.server.access_game_state() as state:
+            with self._client.server.access_game() as state:
                 state.move_down_all_the_way(self.player)
         elif received in (b"R", b"r"):
             self._client.rotate_counter_clockwise = (
                 not self._client.rotate_counter_clockwise
             )
+        # TODO: remove, for development only
```

Comment:
I like this feature and I think it should be documented in README instead of removed.